File: Python3/1059.py
from math import sqrt
MAXN = 100010
n = int(input())
prime = []
fac = []
flag = [False for i in range(MAXN)]
def find_prime():
    for i in range(2,MAXN):
        if flag[i] == False:
            prime.append(i)
            for j in range(2*i,MAXN,i):
                flag[i] = True
find_prime()
if n == 1:
    print("1=1")
else:
    print("%d="%n,end='')
    sqr = int(sqrt(n))
    for i in range(len(prime)):
        if prime[i] > sqr:
            break
        if n % prime[i] == 0:
            factor = prime[i]
            cnt = 0
            while n % prime[i] == 0:
                cnt += 1
                n = n//prime[i]
            fac.append((factor,cnt))
        if n== 1:
            break
    if n != 1:
        fac.append((n,1))
    for each in fac:
        if each != fac[0]:
            print("*",end='')
        print("%d"%each[0],end='')
        if each[1] > 1:
            print("^%d"%each[1],end='')

File: c++/1059.cpp
//
//  main.cpp
//  test
//
//  Created by 李寻欢 on 2019/10/8.
//  Copyright © 2019 李寻欢. All rights reserved.
//
#include <stdio.h>
#include <iostream>
#include <string>
#include <map>
#include <cmath>
#include <algorithm>
#include <stack>
#include <vector>
#include <queue>
#include <unordered_map>
#include <string.h>
#include <set>
using namespace std;

const int MAXN = 1e5+10;
struct factor{
    int x,cnt;
}fac[10];
int prime[MAXN];
int pnum =0 ;
bool is[MAXN] = { false};
void find_prime(){
    for(int i=2;i<MAXN;++i){
        if (is[i] == false) {
            prime[pnum++] = i;
            for(int j=i+i;j<MAXN;j+=i) is[j] = true;
        }
    }
}

int main(){
    find_prime();
    int n,num = 0;
    scanf("%d",&n);
    if(n == 1) printf("1=1");
    else{
        printf("%d=",n);
        int sqr = (int)sqrt(1.0*n);
        for(int i=0;i<pnum && prime[i] <=sqr ;++i){
            if(n % prime[i] == 0){
                fac[num].x = prime[i];
                fac[num].cnt = 0;
                while (n % prime[i] == 0) {
                    fac[num].cnt++;
                    n /= prime[i];
                }
                num++;
            }
            if(n==1)break;
        }
        if(n != 1 ){
            fac[num].x = n;
            fac[num++].cnt = 1;
        }
        for(int i=0;i<num;++i){
            if(i>0) printf("*");
            printf("%d",fac[i].x);
            if(fac[i].cnt > 1)
                printf("^%d",fac[i].cnt);
        }
    }
}


